geomesa-spark/geomesa-spark-sql/src/main/scala/org/apache/spark/sql/SQLRules.scala

```diff
@@ -113,8 +114,78 @@ object SQLRules extends LazyLogging {
       }
     }
 
+    private def extractGridId(envelopes: List[Envelope], e: org.apache.spark.sql.catalyst.expressions.Expression): Option[List[Int]] = e match {
+      case And(l, r) => extractGridId(envelopes, l).orElse(extractGridId(envelopes, r))
+      case ScalaUDF(_, _, Seq(_, GeometryLiteral(_, geom)), _) => Some(RelationUtils.gridIdMapper(geom, envelopes))
+      case GeometryLiteral(_,geom) => Some(RelationUtils.gridIdMapper(geom, envelopes))
+      case _ => None
+    }
+
+    // Converts a pair of GeoMesaRelations into one GeoMesaJoinRelation
+    private def alterRelation(left: GeoMesaRelation, right: GeoMesaRelation, cond: Expression): GeoMesaJoinRelation = {
+      val joinedSchema = StructType(left.schema.fields ++ right.schema.fields)
```

Comment:
```schema.fields ``` is an ```Array[StructField]```. So using `++` should just append to the end.

In my tests there is a common field with the same name and type, and I had no problems. At the Catalyst level, attributes are bound with unique identifiers